### PR TITLE
Custom joy node

### PR DESCRIPTION
**Pull request by Achllle**
This PR introduces a node that allows someone to change parameters in other nodes based on Joy messages. Good starting point to making other modifications. It is disabled by default and I added a section for it in the readme.

Depends #178

### CODE REVIEW

**Comment by Achllle** [comment]
@abust005

**Review by abust005 — CHANGES_REQUESTED** [review]
Looks good, aside from that one comment

File: ROS/osr_control/osr_control/joy_extras.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        
+        self.subscription = self.create_subscription(
+            Joy,
+            'joy',
+            self.joy_callback,
+            10)
+        self.log.info("Initialized JoyButtonSubscriber.")
+
+    async def joy_callback(self, msg):
+        if len(msg.buttons) < self.duty_button_index:
```
[the remark this answers is not in the document]
Comment (Achllle) [reply ROS/osr_control/osr_control/joy_extras.py:33]: you're right, good catch!

[the remark this answers is not in the document]
Comment (Achllle) [reply ROS/osr_control/osr_control/joy_extras.py:33]:
```suggestion
        if len(msg.buttons) <= self.duty_button_index:
```

**Review by apollokit — APPROVED** [review]
Looks fine to me, tho haven't tested. Seems like nice functionality!

Doesn't this overlap with https://github.com/nasa-jpl/osr-rover-code/pull/178 ?

**Review by abust005 — APPROVED** [review]

**Comment by Achllle** [comment]
Yes, I'd added a dependency on #178 in the PR description. Since that's in now, I'll go ahead with merging this one as well

**Pull request merged**